target/linux/ramips/patches-6.6/821-SPI-ralink-add-Ralink-SoC-spi-driver.patch

```diff
@@ -507,50 +496,12 @@ Acked-by: John Crispin <blogic@openwrt.org>
 +	master->set_cs = rt2880_spi_set_cs;
 +	master->transfer_one = rt2880_spi_transfer_one,
 +
-+	dev_set_drvdata(&pdev->dev, master);
-+
 +	rs = spi_master_get_devdata(master);
 +	rs->master = master;
 +	rs->base = base;
 +	rs->clk = clk;
 +
-+	if (atomic_inc_return(&hw_reset_count) == 1) {
-+		ret = device_reset(&pdev->dev);
```

Comment:
I think this code is used to call the `device_reset` only for the first instance of this device. Probably multiple SPI cores  share the reset line and the driver wants to do a HW reset before the first instance is used. 

I think you can call it always, if the reset is shared it is only executed for the first device, see:
https://elixir.bootlin.com/linux/v6.6.53/source/drivers/reset/core.c#L327

Reply:
That's a different function, no?

Reply:
In the end `device_reset` calls `reset_control_reset`. I think you can remove the `hw_reset_count` handling and just always call `device_reset(&pdev->dev);`.